src/main/java/de/eldoria/pickmeup/listener/CarryListener.java

```diff
@@ -92,57 +92,113 @@ public void onEntityInteract(PlayerInteractAtEntityEvent event) {
-                for (Entity passenger : player.getPassengers()) {
-                    delayedActions.schedule(() -> trailHandler.startTrail(passenger), 2);
-                    player.removePassenger(passenger);
-                    passenger.setVelocity(throwVec);
-                    plugin.getLogger().config("Throwing entity | Location:" + player.getLocation().toVector()
-                                              + " | Force: " + force
-                                              + " | ThrowForce: " + config.carrySettings().throwForce()
-                                              + " | ViewVec: " + throwVec);
+    private void performThrow(Player player, UUID playerUUID) {
+        double force = throwBarHandler.getAndRemove(player);
+        Vector throwVec = player.getEyeLocation().getDirection().normalize().multiply(force * config.carrySettings().throwForce());
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_ENDER_DRAGON_FLAP, 1, 1);
+
+        List<Entity> passengers = new ArrayList<>(player.getPassengers());
+        for (Entity passenger : passengers) {
+            if (passenger == null || !passenger.isValid() || passenger.isDead()) {
+                player.removePassenger(passenger);
+                blocked.remove(passenger.getUniqueId());
+                continue;
+            }
+
+            delayedActions.schedule(() -> {
+                if (passenger.isValid() && !passenger.isDead()) {
+                    trailHandler.startTrail(passenger);
                 }
+            }, 2);
+
+            player.removePassenger(passenger);
+            passenger.setVelocity(throwVec);
+
+            plugin.getLogger().config("Throwing entity | Location:" + player.getLocation().toVector()
+                    + " | Force: " + force
+                    + " | ThrowForce: " + config.carrySettings().throwForce()
+                    + " | ViewVec: " + throwVec);
+        }
+        mountStates.remove(playerUUID);
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent event) {
+        Player player = event.getEntity();
+        UUID playerUUID = player.getUniqueId();
+        mountStates.remove(playerUUID);
+        throwBarHandler.remove(player);
+    }
+
+    @EventHandler
+    public void onEntityDeath(EntityDeathEvent event) {
+        LivingEntity entity = event.getEntity();
+        for (Player player : Bukkit.getOnlinePlayers()) {
+            if (player.getPassengers().contains(entity)) {
+                player.removePassenger(entity);
+                mountStates.remove(player.getUniqueId());
+                throwBarHandler.remove(player);
             }
-            mountStates.remove(player.getUniqueId());
         }
     }
 
+    @EventHandler
+    public void onPlayerQuit(PlayerQuitEvent event) {
+        Player player = event.getPlayer();
+        UUID playerUUID = player.getUniqueId();
+        mountStates.remove(playerUUID);
+        throwBarHandler.remove(player);
```

Comment:
Again no need for declaration of single use variables.
```suggestion
        mountStates.remove(event.getPlayer().getUniqueId());
        throwBarHandler.remove(event.getPlayer());
```